File: backend/schemas/debt_report.py
"""Pydantic schemas for debt report operations."""

from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class AgingBucketInvoice(BaseModel):
    """Minimal invoice info for aging bucket."""
    id: int
    invoice_number: str
    created_at: datetime
    total: float
    paid_amount: float
    remaining_amount: float

    class Config:
        from_attributes = True


class AgingBucket(BaseModel):
    """Aging bucket schema."""
    bucket_label: str  # e.g., "0-30 ngày", "30-60 ngày"
    bucket_key: str  # e.g., "0-30", "30-60", "60-90", "90+"
    invoice_count: int
    total_amount: float
    invoices: List[AgingBucketInvoice] = []

    class Config:
        from_attributes = True


class AgingAnalysisResponse(BaseModel):
    """Aging analysis response schema."""
    customer_id: Optional[int] = None  # None means all customers
    customer_name: Optional[str] = None
    buckets: List[AgingBucket]
    total_debt: float
    total_invoices: int
    generated_at: datetime

    class Config:
        from_attributes = True

File: react-frontend/src/types/debtReport.ts
/**
 * Debt report related TypeScript types
 */

export interface AgingBucketInvoice {
  id: number
  invoice_number: string
  created_at: string
  total: number
  paid_amount: number
  remaining_amount: number
}

export interface AgingBucket {
  bucket_label: string  // "0-30 ngày", "30-60 ngày"
  bucket_key: string    // "0-30", "30-60", "60-90", "90+"
  invoice_count: number
  total_amount: number
  invoices: AgingBucketInvoice[]
}

export interface AgingAnalysisResponse {
  customer_id?: number | null
  customer_name?: string | null
  buckets: AgingBucket[]
  total_debt: number
  total_invoices: number
  generated_at: string
}
